Add rustfmt and modify Rust a little

nrc opened the pull request:
Requires https://github.com/mozilla/dxr/pull/503

nrc commented on config.yml:
```diff
@@ -109,12 +109,30 @@ trees:
+    es_index: 'dxr_hot_{format}_{tree}_{unique}'
+    enabled_plugins: 'pygmentize rust buglink'
+    ignore_patterns: '/target/ /tests/ /data/ .hg .git CVS .svn .bzr .deps .libs .DS_Store .nfs* *~ ._* *.pyc'
+    build_command: 'cd $source_folder && RUSTC=/builds/dxr-build-env/dxr/dxr/plugins/rust/dxr-rustc.sh cargo build --verbose'
```
Is this the right directory? It's the best I could figure out from the config scripts

nrc commented on config.yml:
```diff
@@ -109,12 +109,30 @@ trees:
+      - url: 'https://github.com/rust-lang-nursery/rustfmt.git'
+    schedule: 'H H/6 * * *'
+    quiet_period: 14400
+    job_weight: 4
```
I'm not sure what these mean - it would be totally fine to lower the priority on Rustfmt and to build it less often (it's not a very big project). I just copied the Rust values to start with.

klibby replied:
proj_dir can be left out; it's use case is when you're indexing a bunch of repos under a parent directory (the proj_dir).  we can adjust the timing as we go; fast builds can run more often, and lean builds (low RAM usage) can get a lower weight.

nrc commented:
I tried to run `deploy update-config` to generate the other files, but it errored out.

klibby commented:
Fixed the issue running `deploy update-config` in 12cf580; thanks for tripping over that!

klibby commented:
#11 will need to be rolled in

klibby commented:
Tried a test build and run into the following:

Starting tree 'rustfmt'.
Running pre_build  [####################################] 100%
rust-dxr environment
Building tree
    Updating registry `https://github.com/rust-lang/crates.io-index`
failed to fetch `https://github.com/rust-lang/crates.io-index`

@nrc do you know what cargo is trying to do there? if I run the source_command line outside of dxr it builds, so maybe it's trying to write somewhere that it can't/shouldn't?

nrc commented:
Rebased, addressed proj_dir thing, and ran deploy.

re the error, I have not come across it before. Cargo does try and put some configuration stuff in `~/.cargo` (or something like that). If that is causing the problem, then you can change the directory with the `CARGO_HOME` environment variable, hopefully that will fix it but I can't experiment since I can't reproduce the problem.

klibby commented:
looks like the issues I saw are wholly to do with my local setup. did a test run on one of the prod builders and it had no issues. it's super fast, so we can stick with the defaults for  job_weight, quiet_period, and schedule.

klibby commented:
It looks like it is creating a .cargo directory under src/rustfmt and then trying to index it and throwing errors; should we be indexing that?

Indexing files     [####--------------------------------]  11%A worker failed while indexing /builds/dxr-build-env/src/rustfmt/.cargo/registry/index/github.com-0a35038f75765ae4/bi/p_/bip_utracker:
Traceback (most recent call last):
  File "/builds/dxr-build-env/dxr/dxr/build.py", line 581, in index_chunk
    index_file(tree, tree_indexers, path, es, index)
  File "/builds/dxr-build-env/dxr/dxr/build.py", line 551, in index_file
    for chunk in bulk_chunks(docs(), docs_per_chunk=300, bytes_per_chunk=10000):
  File "/builds/dxr-build-env/venv/local/lib/python2.7/site-packages/pyelasticsearch/utils.py", line 31, in bulk_chunks
    for action in actions:
  File "/builds/dxr-build-env/dxr/dxr/build.py", line 516, in docs
    chain.from_iterable(linkses)]
  File "/builds/dxr-build-env/dxr/dxr/plugins/omniglot.py", line 27, in items
    yield 'log', "Log", self.vcs.generate_log(vcs_relative_path)
  File "/builds/dxr-build-env/dxr/dxr/vcs.py", line 216, in generate_log
    return self.upstream + "/commits/" + self.revision + "/" + path
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'

nrc commented:
Hmm, I didn't get these errors when I ran it locally. There also seems to be a lot of missing info. I'll try to repro here...
